gdpr/impl.go

```diff
@@ -128,14 +128,18 @@ func (p *permissionsImpl) allowSync(ctx context.Context, vendorID uint16, consen
 }
 
 func (p *permissionsImpl) allowActivities(ctx context.Context, vendorID uint16, consent string, weakVendorEnforcement bool) (allowBidRequest bool, passGeo bool, passID bool, err error) {
-	parsedConsent, vendor, err := p.parseVendor(ctx, vendorID, consent)
+	parsedConsent, vendor, err := p.parseVendor(ctx, vendorID, consent, weakVendorEnforcement)
 	if err != nil {
 		return false, false, false, err
 	}
 
 	// vendor will be nil if not a valid TCF2 consent string
 	if vendor == nil {
-		return false, false, false, nil
+		if weakVendorEnforcement {
```

Comment:
I think this will now allow a TCF1 consent string to be processed if `weakVendorEnforcement` is set to true. `parseVendor` returns with a nil vendor and a nil error when the consent string is considered a well-formed TCF1 or TCF2 string with a version != 2. That caused it to enter this `if` block which would return `false, false, false, nil`. See `TestTCF1Consent`.

Perhaps `parseVendor` should return an error when `version != 2`. Then a valid TCF1 consent string would be handled by the err check on line 132 and you could keep your logic changes here. This change would also mean a version != 2 would be handled the same way as a malformed consent string. Thoughts?

Reply:
Good idea, but moved the verification slightly.